tools/sigma/backends/uberagent.py

```diff
@@ -0,0 +1,371 @@
+
+    def __str__(self):
+        """Builds and returns the [ActivityMonitoringRule] configuration block."""
+        result = "[ActivityMonitoringRule]\n"
+
+        # The Description is optional.
+        if len(self.description) > 0:
+            result += "# {}\n".format(self.description)
+
+        # Make sure all required properties have at least a value that is somehow usable.
+        if self.event_type is None:
+            raise MalformedRuleException()
+
+        if len(self.tag) == 0:
+            raise MalformedRuleException()
+
+        if len(self.name) == 0:
+            raise MalformedRuleException()
+
+        if len(self.query) == 0:
+            raise MalformedRuleException()
+
+        result += "RuleName = {}\n".format(self.name)
+        result += "EventType = {}\n".format(self.event_type)
+        result += "Tag = {}\n".format(self._prefixed_tag())
+
+        # The RiskScore is optional.
+        # Set it, if a risk_score value is present.
+        if self.risk_score > 0:
+            result += "RiskScore = {}\n".format(self.risk_score)
+
+        result += "Query = {}\n".format(self.query)
+        return result
+
+
+def get_parser_properties(sigmaparser):
+    title = sigmaparser.parsedyaml['title']
+    level = sigmaparser.parsedyaml['level']
+    description = sigmaparser.parsedyaml['description']
+    condition = sigmaparser.parsedyaml['detection']['condition']
+    logsource = sigmaparser.parsedyaml['logsource']
+    category = ''
+    if 'category' in logsource:
+        category = logsource['category'].lower()
+
+    product = ''
+    if 'product' in logsource:
+        product = logsource['product'].lower()
+
+    service = ''
+    if 'service' in logsource:
+        service = logsource['service'].lower()
+
+    return product, category, service, title, level, condition, description
+
+
+def write_file_header(f, level):
+    f.write("#\n")
+    f.write("# The rules are generated from the Sigma GitHub repository at https://github.com/Neo23x0/sigma\n")
+    f.write("# Follow these steps to get the latest rules from the repository with Python\n")
+    f.write("#    1. Clone the repository locally\n")
+    f.write("#    2. Using a commandline, change working directory to the just cloned repository\n")
+    f.write("#    3. Run sigmac -I --target uberagent -r rules/\n")
+    f.write("#\n")
+    f.write("# The rules in this file are marked with sigma-level: {}\n".format(level))
+    f.write("#\n\n")
+
+
+class uberAgentBackend(SingleTextQueryBackend):
+    """Converts Sigma rule into uberAgent ESA's process tagging rules."""
+    identifier = "uberagent"
+    active = True
+    config_required = False
+    rule = None
+
+    #
+    # SingleTextQueryBackend
+    #
+    andToken = " and "
+    orToken = " or "
+    notToken = "not "
+    subExpression = "(%s)"
+    listExpression = "[%s]"
+    listSeparator = ", "
+    valueExpression = "\"%s\""
+    nullExpression = "is null"
```

Comment:
Sollte hier nicht auch ein Leerzeichen vor die Schlüsselwörter? (genauso bei `is not null`